src/main/java/org/opentripplanner/graph_builder/GraphBuilder.java

```diff
@@ -133,6 +137,11 @@ public static GraphBuilder create(
             pruneFloatingIslands.setPruningThresholdIslandWithStops(config.pruningThresholdIslandWithStops);
             graphBuilder.addModule(pruneFloatingIslands);
         }
+        if (hasTransitData) {
+            SubmodesConfiguration submodesConfiguration = new SubmodesConfiguration();
+            submodesConfiguration.setConfig(submodesConfig);
+            graphBuilder.addModule(submodesConfiguration);
```

Comment:
This should be moved to the OTPAppConstruction class and created as a bean there. I will add a comment there.

``

Reply:
I have renamed SubmodesConfiguration to TransitModeServiceModule. I'm not sure it belongs in the OTPAppConstruction, since it is a property of the graph and not that particular OTP instance.